GymBuddies/ViewControllers/AddExerciseViewController.m

```diff
@@ -0,0 +1,92 @@
+//
+//  AddExerciseViewController.m
+//  GymBuddies
+//
+//  Created by Gael Rodriguez Gomez on 7/11/22.
+//
+
+#import "AddExerciseViewController.h"
+#import "ParseAPIManager.h"
+#import "AddExerciseTableViewCell.h"
+#import "SavedExercise.h"
+#import "CreateExerciseViewController.h"
+#import "AlertCreator.h"
+
+
+@interface AddExerciseViewController () <UITableViewDelegate, UITableViewDataSource, CreateExerciseViewControllerDelegate>
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *exercises;
+
+@end
+
+@implementation AddExerciseViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+    self.exercises = [[NSMutableArray alloc]init];
+    [self loadUsersExercises];
+}
+
+
+#pragma mark - Data fetching
+
+-(void)loadUsersExercises{
+    [ParseAPIManager fetchUsersExercises:^(NSArray * _Nonnull elements, NSError * _Nonnull error) {
+        if(elements!=nil){
+            for(SavedExercise *element in elements){
+                [self.exercises addObject:element.exercise];
```

Comment:
should you consider deduping here? what if there are exercises in the new ones you are adding that overlap with existing routines?